api/plugin_helpers.go

```diff
@@ -102,7 +103,7 @@ func VaultPluginTLSProviderContext(ctx context.Context, apiTLSConfig *TLSConfig)
 	return func() (*tls.Config, error) {
 		unwrapToken := os.Getenv(PluginUnwrapTokenEnv)
 
-		parsedJWT, err := jwt.ParseSigned(unwrapToken)
+		parsedJWT, err := jwt.ParseSigned(unwrapToken, []jose.SignatureAlgorithm{jose.ES256})
```

Comment:
Shouldn't this be `ES512`?

Also, maybe we should have an exported constant for the signing algorithm we support, and `unwrapToken` can use that, so we don't accidentally get out of sync in the future?

Reply:
Sorry yeah this is a weird change, and I only briefly touched on it in the PR description. I just wanted to see the CI fail with this and pass with `ES512` for extra confidence that there is genuine test coverage already.

A shared constant seems like a good idea, although I'd need to put it in the API package to make sure the API package doesn't need to depend on the Vault package. And while that would be fine, this code is truthfully in the wrong package, it should be in the SDK, and it's also legacy/backwards compatibility code because plugins should now all use go-plugin's auto-TLS feature, so I'm slightly hesitant to broaden the exported footprint of this code :/ WDYT?

Reply:
It's not ideal, but it doesn't seem too bad to have something like `SupportedJwtSignatureAlgorithm` be an `api/` exported constant and having `unwrapToken` use that?

The Vault core already depends on the `api/` package, so it should be safe enough.